tests/std/tests/P1682R3_to_underlying/test.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright(c) Microsoft Corporation.
+// SPDX - License - Identifier : Apache - 2.0 WITH LLVM - exception
+
+#include <cassert>
+#include <type_traits>
+#include <utility>
+
+using namespace std;
+
+// TRANSITION, GH-602
+// template <class T>
+// constexpr bool is_valid = requires(T e) { to_underlying(e); };
+
+template <class T, class = void>
+struct can_underlying : false_type {};
+template <class T>
+struct can_underlying<T, void_t<decltype(to_underlying(declval<T>()))>> : true_type {};
+
+template <class T>
+constexpr bool is_valid = can_underlying<T>::value;
+
+enum enum1 : char { a = '1' };
+enum class enum2 : int { b = 2 };
+enum class enum3 : long { c = 3 };
+

```

Comment:
I'd also like to see a test case for a scoped enumeration with no explicit underlying type a la `enum class enum4 { d = 4 };` (whose underlying type is implicitly `int`).